From 454ed79452c6ea33a62ae4c9f34188027167dff8 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 15 Feb 2021 10:25:35 +0000
Subject: [PATCH 338/552] gpio-fsm: Rename 'num-soft-gpios' to avoid warning

As of 5.10, the Device Tree parser warns about properties that look
like references to "suppliers" of various services. "num-soft-gpios"
resembles a declaration of a GPIO called "num-soft", causing the value
to be interpreted as a phandle, the owner of which is checked for a
"#gpio-cells" property.

To avoid this warning, rename the gpio-fsm property to "num-swgpios".

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/gpio/gpio-fsm.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpio/gpio-fsm.c b/drivers/gpio/gpio-fsm.c
index 2cab50e51315..708f258fda35 100644
--- a/drivers/gpio/gpio-fsm.c
+++ b/drivers/gpio/gpio-fsm.c
@@ -866,8 +866,9 @@ static int gpio_fsm_probe(struct platform_device *pdev)
 		[SYM_SHUTDOWN] = "shutdown_state",
 	};
 
-	if (of_property_read_u32(np, "num-soft-gpios", &num_soft_gpios)) {
-		dev_err(dev, "missing 'num-soft-gpios' property\n");
+	if (of_property_read_u32(np, "num-swgpios", &num_soft_gpios) &&
+	    of_property_read_u32(np, "num-soft-gpios", &num_soft_gpios)) {
+		dev_err(dev, "missing 'num-swgpios' property\n");
 		return -EINVAL;
 	}
 
-- 
2.33.0

